newrelic/resource_newrelic_workload.go

```diff
@@ -0,0 +1,327 @@
+package newrelic
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/errors"
+	"github.com/newrelic/newrelic-client-go/pkg/workloads"
+)
+
+func resourceNewRelicWorkload() *schema.Resource {
+	userReference := &schema.Schema{
+		Type:     schema.TypeList,
+		MaxItems: 1,
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"email": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"gravatar": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"id": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+				"name": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceNewRelicWorkloadCreate,
+		Read:   resourceNewRelicWorkloadRead,
+		Update: resourceNewRelicWorkloadUpdate,
+		Delete: resourceNewRelicWorkloadDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"account": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
```

Comment:
Should this be Optional/Computed? It seems like it's necessary to build this [resource](https://github.com/terraform-providers/terraform-provider-newrelic/pull/474/files#diff-d8b554046f06d5baab974aeb013d168aR155). If this isn't required, we'll have to tune the logic in that link to reflect that.